GenerateOppMessage.cmake

```diff
@@ -1,31 +1,99 @@
 include(CMakeParseArguments)
 
+# Options:
+# - MSG4: Call message compiler with --msg4
+# 
+# One Value Arguments:
+# - TARGET: Optional name of the target to call target_sources and target_include_directories to add compiled sources
+# - OUTPUT_ROOT: Optional root-directory where the files will be generated. Default: ${PROJECT_BINARY_DIR}/opp_messages
+# - DIRECTORY: Optional relative path for the output-directory (Appended to OUTPUT_ROOT). 
+#   Defaults to the relative path of the message file in respect to ${PROJECT_SOURCE_DIR}/src
+# - SRCS: Optional name of the variable to populate with sources to be compiled
```

Comment:
What is the more common name in CMake, *SRCS* or *SOURCES*? I know that *FindProtobuf.cmake* names its (positional) parameter *SRCS*. I just want to keep it consistent with other CMake code.

Reply:
Have seen SRCS a lot. @thor - your thoughts?

Reply:
I've seen `SRCS` too. On the other hand, `SOURCES` is also a [target property](https://cmake.org/cmake/help/latest/prop_tgt/SOURCES.html).

As long as we're consistent internally I'm all 👍, though on a non-CMake-related matter: I prefer fully spelled out variables `commands` over `cmds`, `table` over `tbl`, `directory` over `dir` (except in a maths related context, but that's different). Thus, in that sense I'd suggest as clear and straightforward variable names as possible, and prefer `SOURCES`.


(That being said, I'm sure I haven't always followed that preference myself, hehe.)

Reply:
The more I think about it, the more I am afraid that users may confuse "SRCS"/"SOURCES" with input sources.
But that might just be me getting paranoid…

Reply:
we could name the variables `GEN_SOURCES` and `GEN_INCLUDE_DIR`?